lte/cloud/go/services/subscriberdb/storage/subscriber_state_test.go

```diff
@@ -101,4 +105,82 @@ func TestSubscriberStorage(t *testing.T) {
 		err = s.DeleteSubscribersForGateway(nid1, gwid2)
 		assert.NoError(t, err)
 	})
+
+	t.Run("Test getting subscriber states by imsi and getting all subscribers", func(t *testing.T) {
+
+		imsisGw1 := []string{"IMSI001010000000123", "IMSI001010000000456"}
+		imsisGw2 := []string{"IMSI002020000000123", "IMSI002020000000456"}
+		subscriberStatesGw1 := CreateTestGatewaySubscriberState(imsisGw1...)
+		subscriberStatesGw2 := CreateTestGatewaySubscriberState(imsisGw2...)
+
+		var frozenTime int64 = 100000
+		err = s.(*subscriberStorage).setTestGatewaySubscriberStateWithIncreasingTimestamp(t, nid1, gwid1, &subscriberStatesGw1, frozenTime)
+		assert.NoError(t, err)
+		err = s.(*subscriberStorage).setTestGatewaySubscriberStateWithIncreasingTimestamp(t, nid1, gwid2, &subscriberStatesGw2, frozenTime)
+		assert.NoError(t, err)
+		err = s.(*subscriberStorage).setTestGatewaySubscriberStateWithIncreasingTimestamp(t, nid2, gwid1, &subscriberStatesGw1, frozenTime)
+		assert.NoError(t, err)
+
+		// get all IMSIs for network
+
+		foundStates, err := s.GetSubscribersForIMSIs(nid1, nil)
+		assert.NoError(t, err)
+		var expectedStates = map[string]state.ArbitraryJSON{}
+		maps.Copy(expectedStates, subscriberStatesGw1.Subscribers)
+		maps.Copy(expectedStates, subscriberStatesGw2.Subscribers)
+		assert.True(t, cmp.Equal(foundStates, expectedStates))
+
+		// get selected IMSIs for network (from two gateways)
+
+		selectedIMSIs := []string{"IMSI001010000000123", "IMSI002020000000456"}
+		expectedStates = map[string]state.ArbitraryJSON{
+			"IMSI001010000000123": subscriberStatesGw1.Subscribers["IMSI001010000000123"],
+			"IMSI002020000000456": subscriberStatesGw2.Subscribers["IMSI002020000000456"],
+		}
+		foundStates, err = s.GetSubscribersForIMSIs(nid1, selectedIMSIs)
+		assert.NoError(t, err)
+		assert.True(t, cmp.Equal(foundStates, expectedStates))
+
+		// get latest entry for duplicate IMSI, appearing in two gateways
+
+		imsisGw2 = []string{"IMSI001010000000123", "IMSI002020000000456"}
+		subscriberStatesGw2 = CreateTestGatewaySubscriberState(imsisGw2...)
+		subscriberStatesGw2.Subscribers["IMSI001010000000123"]["magma.ipv4"].([]interface{})[0].(map[string]interface{})["active_duration_sec"] = float64(40)
+		subscriberStatesGw2.Subscribers["IMSI001010000000123"]["magma.ipv4"].([]interface{})[0].(map[string]interface{})["ipv4"] = "10.2.1.12"
+
+		frozenTime = 200000
+		err = s.(*subscriberStorage).setTestGatewaySubscriberStateWithIncreasingTimestamp(t, nid1, gwid2, &subscriberStatesGw2, frozenTime)
+		assert.NoError(t, err)
+		foundStates, err = s.GetSubscribersForIMSIs(nid1, selectedIMSIs)
+		assert.NoError(t, err)
+		assert.True(t, cmp.Equal(foundStates, subscriberStatesGw2.Subscribers))
+	})
+}
+
+func (ss *subscriberStorage) setTestGatewaySubscriberStateWithIncreasingTimestamp(t *testing.T, networkID string, gatewayID string, subscriberStates *GatewaySubscriberState, frozenTime int64) error {
```

Comment:
I think the name of this function is based on an old version, don't see the increasing timestamp any more.